src/idc/zephyr_idc.c

```diff
@@ -120,6 +129,18 @@ int idc_send_msg(struct idc_msg *msg, uint32_t mode)
 	int ret;
 	int idc_send_memcpy_err __unused;
 
+	if (!cpu_is_core_enabled(target_cpu)) {
+		tr_err(&zephyr_idc_tr, "Core %u is down, cannot sent IDC message", target_cpu);
+		return -EACCES;
+	}
+
+	if (zmsg->work_status == WORK_SUBMITTED) {
+		work->sync = 1;
+		ret = k_p4wq_wait(work, K_USEC(IDC_TIMEOUT));
```

Comment:
This is bit of double-counting, the internal semaphore in p4wq should in theory be sufficient. The only case where this matters is the very first usage. The work item is zeroed out at init, so k_p4wq_wait() on uninitialized work item is undefined behaviour, so we need some outside bookeeping. Let me see if I can clean this up further.

Reply:
Droped this code in V2 of the patchset.